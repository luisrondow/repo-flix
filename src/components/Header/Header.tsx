import { NavLink } from 'react-router-dom'
import Text from '../Text/Text'
import useAuth from '../../hooks/useAuth'

const Header = (props: { currentPage: string; username: string }) => {
  const { currentPage, username } = props

  const { logout } = useAuth()

  const handleLogout = () => {
    logout()
  }

  return (
    <header className="flex h-24 w-full items-center bg-base px-4 sm:px-16">
      <div className="flex h-full w-6/12 items-center sm:space-x-12">
        <NavLink to="/">
          <Text as="h1">LOGO</Text>
        </NavLink>
        <NavLink to="/">
          <Text
            as="h2"
            className={`${currentPage === 'home' && 'underline'} ml-4 cursor-pointer sm:ml-0`}
          >
            Discovery
          </Text>
        </NavLink>
      </div>
      <div className="flex h-full w-6/12 items-center justify-end space-x-4 sm:space-x-12">
        <NavLink to="/profile">
          <Text as="h2" className={`${currentPage === 'profile' && 'underline'} cursor-pointer`}>
            {username}
          </Text>
        </NavLink>
        <button onClick={handleLogout}>
          <Text as="h2" className="cursor-pointer">
            Logout
          </Text>
        </button>
      </div>
    </header>
  )
}

export default Header
