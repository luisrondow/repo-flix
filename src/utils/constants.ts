export const TECHS = ['Typescript', 'Javascript', 'Vue', 'Go', 'CSS', 'Node'] as const

export const BOOKMARKS_KEY = 'repoflix:bookmarks' as const
export const SELECTED_TECHS_KEY = 'repoflix:selected-techs' as const
