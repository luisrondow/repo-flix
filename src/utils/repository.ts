export type Repository = {
  id: string
  name: string
  url: string
  image: string
}
