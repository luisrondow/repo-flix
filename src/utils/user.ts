export type User = {
  uid: string
  email: string
  accessToken: string
  username: string
}
