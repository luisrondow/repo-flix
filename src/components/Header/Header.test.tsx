import Header from './Header'
import { renderWithRouter } from '../../utils/renderWithRouter'

describe('Header', () => {
  const props = {
    currentPage: 'home',
    username: 'Username',
  }

  it('should render correctly', () => {
    const { container } = renderWithRouter(
      <Header currentPage={props.currentPage} username={props.username} />,
    )

    expect(container).toHaveTextContent('LOGO')
    expect(container).toHaveTextContent('Discovery')
    expect(container).toHaveTextContent('Username')
    expect(container).toHaveTextContent('Logout')

    expect(container).toMatchSnapshot()
  })

  it('should render correctly the nav links', () => {
    const { container, getByText } = renderWithRouter(
      <Header currentPage={props.currentPage} username={props.username} />,
    )

    const logo = getByText('LOGO')
    const discovery = getByText('Discovery')
    const username = getByText('Username')

    expect(logo).toBeInTheDocument()
    expect(logo.closest('a')).toHaveAttribute('href', '/')

    expect(discovery).toBeInTheDocument()
    expect(discovery.closest('a')).toHaveAttribute('href', '/')

    expect(username).toBeInTheDocument()
    expect(username.closest('a')).toHaveAttribute('href', '/profile')

    expect(container).toMatchSnapshot()
  })

  it('should render correctly the logout button', () => {
    const { container, getByText } = renderWithRouter(
      <Header currentPage={props.currentPage} username={props.username} />,
    )

    const logout = getByText('Logout')

    // todo: test the click event when auth is implemented
    expect(logout).toBeInTheDocument()
    expect(logout.closest('button')).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})
