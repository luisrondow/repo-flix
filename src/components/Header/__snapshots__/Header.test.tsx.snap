// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div>
  <header
    class="flex h-24 w-full items-center bg-base px-4 sm:px-16"
  >
    <div
      class="flex h-full w-6/12 items-center sm:space-x-12"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <h1
          class=" text-[#e0def4] font-roboto text-2xl font-bold sm:text-4xl"
        >
          LOGO
        </h1>
      </a>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <h2
          class="underline ml-4 cursor-pointer sm:ml-0 text-[#e0def4] font-roboto font-bold sm:text-xl"
        >
          Discovery
        </h2>
      </a>
    </div>
    <div
      class="flex h-full w-6/12 items-center justify-end space-x-4 sm:space-x-12"
    >
      <a
        class=""
        href="/profile"
      >
        <h2
          class="false cursor-pointer text-[#e0def4] font-roboto font-bold sm:text-xl"
        >
          Username
        </h2>
      </a>
      <button>
        <h2
          class="cursor-pointer text-[#e0def4] font-roboto font-bold sm:text-xl"
        >
          Logout
        </h2>
      </button>
    </div>
  </header>
</div>
`;

exports[`Header should render correctly the logout button 1`] = `
<div>
  <header
    class="flex h-24 w-full items-center bg-base px-4 sm:px-16"
  >
    <div
      class="flex h-full w-6/12 items-center sm:space-x-12"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <h1
          class=" text-[#e0def4] font-roboto text-2xl font-bold sm:text-4xl"
        >
          LOGO
        </h1>
      </a>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <h2
          class="underline ml-4 cursor-pointer sm:ml-0 text-[#e0def4] font-roboto font-bold sm:text-xl"
        >
          Discovery
        </h2>
      </a>
    </div>
    <div
      class="flex h-full w-6/12 items-center justify-end space-x-4 sm:space-x-12"
    >
      <a
        class=""
        href="/profile"
      >
        <h2
          class="false cursor-pointer text-[#e0def4] font-roboto font-bold sm:text-xl"
        >
          Username
        </h2>
      </a>
      <button>
        <h2
          class="cursor-pointer text-[#e0def4] font-roboto font-bold sm:text-xl"
        >
          Logout
        </h2>
      </button>
    </div>
  </header>
</div>
`;

exports[`Header should render correctly the nav links 1`] = `
<div>
  <header
    class="flex h-24 w-full items-center bg-base px-4 sm:px-16"
  >
    <div
      class="flex h-full w-6/12 items-center sm:space-x-12"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <h1
          class=" text-[#e0def4] font-roboto text-2xl font-bold sm:text-4xl"
        >
          LOGO
        </h1>
      </a>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <h2
          class="underline ml-4 cursor-pointer sm:ml-0 text-[#e0def4] font-roboto font-bold sm:text-xl"
        >
          Discovery
        </h2>
      </a>
    </div>
    <div
      class="flex h-full w-6/12 items-center justify-end space-x-4 sm:space-x-12"
    >
      <a
        class=""
        href="/profile"
      >
        <h2
          class="false cursor-pointer text-[#e0def4] font-roboto font-bold sm:text-xl"
        >
          Username
        </h2>
      </a>
      <button>
        <h2
          class="cursor-pointer text-[#e0def4] font-roboto font-bold sm:text-xl"
        >
          Logout
        </h2>
      </button>
    </div>
  </header>
</div>
`;
